fabric_generator/fabric_gen.py

```diff
@@ -15,6380 +15,2278 @@
+                # This was done such that the index from left-to-right in the adjacency matrix corresponds with the multiplexer select input (index)
+                numGnd = 0
+                muxComponentName = ""
+                if (self.fabric.multiplexerStyle == MultiplexerStyle.CUSTOM) and (muxSize == 2):
+                    muxComponentName = 'my_mux2'
+                elif (self.fabric.multiplexerStyle == MultiplexerStyle.CUSTOM) and (2 < muxSize <= 4):
+                    muxComponentName = 'cus_mux41_buf'
+                    numGnd = 4-muxSize
+                elif (self.fabric.multiplexerStyle == MultiplexerStyle.CUSTOM) and (4 < muxSize <= 8):
+                    muxComponentName = 'cus_mux81_buf'
+                    numGnd = 8-muxSize
+                elif (self.fabric.multiplexerStyle == MultiplexerStyle.CUSTOM) and (8 < muxSize <= 16):
+                    muxComponentName = 'cus_mux161_buf'
+                    numGnd = 16-muxSize
+
+                portList, signalList = [], []
+                start = 0
+                for start in range(muxSize):
+                    portList.append(f"A{start}")
+                    signalList.append(f"{portName}_input[{start}]")
+
+                for end in range(start, numGnd):
+                    portList.append(f"A{end}")
+                    signalList.append(f"GND0")
+
+                if self.fabric.multiplexerStyle == MultiplexerStyle.CUSTOM:
+                    if muxSize == 2:
+                        portList.append(f"S")
+                        signalList.append(
+                            f"ConfigBits[{old_ConfigBitstreamPosition}+0]")
                     else:
-                        ConfigBitsOrder.append(int(k))
-            elif RangeItem.isdigit():
-                if int(RangeItem) in ConfigBitsOrder:
-                    raise ValueError('Configuration bit index '+str(RangeItem)+' already allocated in ', entity, line[frame_name])
+                        for i in range(int(math.ceil(math.log2(muxSize)))):
+                            portList.append(f"S{i}")
+                            portList.append(f"S{i}N")
+                            signalList.append(
+                                f"ConfigBits[{old_ConfigBitstreamPosition}+{i}]")
+                            signalList.append(
+                                f"ConfigBits[{old_ConfigBitstreamPosition}+{i}]")
+
+                portList.append("X")
+                signalList.append(f"{portName}")
+
+                if (self.fabric.multiplexerStyle == MultiplexerStyle.CUSTOM):
+                    self.writer.addAssignScalar(
+                        f"{portName}_input", connections[portName][::-1], int(self.fabric.generateDelayInSwitchMatrix))
+                    self.writer.addInstantiation(compName=muxComponentName,
+                                                 compInsName=f"inst_{muxComponentName}",
```

Comment:
this needs to be creating unique names for each instance, you can't have two instances in a module with the same name